drivers/src/dma.rs

```diff
@@ -679,18 +680,19 @@ impl<'a> DmaRecovery<'a> {
 
         #[cfg(any(feature = "fpga_realtime", feature = "fpga_subsystem"))]
         {
-            // FPGA implementation: wait for FIFO to be not empty and read dword by dword
+            // FPGA implementation: wait for FIFO to be full every 256 bytes
             let len = (image_size_bytes as usize / 4).min(buffer.len());
             for i in 0..len {
-                // Wait for FIFO to not be empty before each read_dword.
-                self.with_regs(|r| {
-                    while r
-                        .sec_fw_recovery_if()
-                        .indirect_fifo_status_0()
-                        .read()
-                        .empty()
-                    {}
-                })?;
+                if i % (INDIRECT_FIFO_SIZE as usize / 4) == 0 {
+                    self.with_regs(|r| {
+                        while !r
+                            .sec_fw_recovery_if()
+                            .indirect_fifo_status_0()
+                            .read()
+                            .full()
+                        {}
+                    })?;
+                }
```

Comment:
Does this not make it necessary to have images with sizes multiple of 256? Let's see what the CI says about this

Reply:
Correct.